api/repository/sprots_sql.go

```diff
@@ -0,0 +1,50 @@
+package repository
+
+import (
+	"context"
+
+	"sports-day/api/db_model"
+	"sports-day/api/pkg/errors"
+
+	"gorm.io/gorm"
+)
+
+type sports struct{}
+
+func NewSports() Sports {
+	return sports{}
+}
+
+func (r sports) Get(ctx context.Context, db *gorm.DB, id string) (*db_model.Sport, error) {
+	var sport db_model.Sport
+	if err := db.First(&sport, "id = ?", id).Error; err != nil {
+		return nil, errors.Wrap(err)
+	}
+	return &sport, nil
+}
+
+func (r sports) List(ctx context.Context, db *gorm.DB) ([]*db_model.Sport, error) {
+	var sports []*db_model.Sport
+	if err := db.Find(&sports).Error; err != nil {
+		return nil, errors.Wrap(err)
+	}
+	return sports, nil
+}
+
+func (r sports) Save(ctx context.Context, db *gorm.DB, sport *db_model.Sport) (*db_model.Sport, error) {
+	if err := db.Save(sport).Error; err != nil {
+		return nil, errors.Wrap(err)
+	}
+	return sport, nil
+}
+
+func (r sports) Delete(ctx context.Context, db *gorm.DB, id string) (*db_model.Sport, error) {
+	var sport db_model.Sport
+	if err := db.First(&sport, "id = ?", id).Error; err != nil {
+		return nil, errors.Wrap(err)
+	}
```

Comment:
エラーハンドリングを強化できるとなお良しです🙆

```suggestion
if errors.Is(err, gorm.ErrRecordNotFound) {
	return nil, errors.ErrUserNotFound
}
```